sys/ztimer/rtc.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+/**
+ * @ingroup     sys_ztimer_rtc
+ * @{
+ *
+ * @file
+ * @brief       ztimer periph/rtc backend implementation
+ *
+ * This implementation simply converts an integer time to split RTC values and
+ * back, which is rather inefficient. If available, use ztimer_rtt...
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+#include "periph/rtc.h"
+#include "ztimer/rtc.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static uint32_t _gmt_civil_to_timestamp(unsigned y, unsigned m, unsigned d,
+                                        unsigned h, unsigned mi, unsigned s)
+{
+    if (m <= 2) {
+        y -= 1;
+        m += 12;
+    }
+    return (365 * y + y / 4 - y / 100 + y / 400 + 3 * (m + 1) / 5 + 30 * m + d -
+            719561) * 86400 + 3600 * h + 60 * mi + s;
+}
+
+void _timestamp_to_gmt_civil(struct tm *_tm, uint32_t epoch)
+{
+    uint32_t s = epoch % 86400;
+
+    epoch /= 86400;
+    uint32_t h = s / 3600;
+    uint32_t m = s / 60 % 60;
+    s = s % 60;
+    uint32_t x = (epoch * 4 + 102032) / 146097 + 15;
+    uint32_t b = epoch + 2442113 + x - (x / 4);
+    uint32_t c = (b * 20 - 2442) / 7305;
+    uint32_t d = b - 365 * c - c / 4;
+    uint32_t e = d * 1000 / 30601;
+    uint32_t f = d - e * 30 - e * 601 / 1000;
```

Comment:
Could you create or use some defines for the magic numbers here?

Reply:
Uff, that's a public domain algorithm. I found it with the note "The algorithmic solution above is part of the programming folklore.". Honestly I don't want to touch it at all.

Reply:
How about adding that quote as a comment above/below the code then?

Reply:
done